### PR TITLE
Abstract classes

### PR DESCRIPTION
**Pull request by SamuelllG**
implemented abstract classes Movable and Damageable

We should check this really properly, since they were issues with the folder structure, so that merging didn't show me the conflicts correctly. I fixed it manually, after having matched the folder structure on my local branch before merge. There shouldn't be issues with stuff that I didn't touch in my branch. For you branch @DavidPgl I just added the field targetfinder in enemycontroller, updated the Update method and dragged the TargetFinder gameobject into the free slot in the enemycontroller.

If this merge is too unsure, we can also close this and do it differently. For me everything works in my local. (I think I also pushed the gamescene, which is a thing to be checked, too) 

Maybe here it is good that you checkout the branch and try it on your own.

### CODE REVIEW

**Review by inkompotato — DISMISSED** [review]
looks like everything is working fine, lets hope merging does not break master...

File: Assets/Scripts/AbstractClasses/Damageable.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using UnityEngine;
+
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/AbstractClasses/Damageable.cs:2]: Please add the correct namespace for the file: `namespace AbstractClasses`

File: Assets/Scripts/AbstractClasses/Movable.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/AbstractClasses/Movable.cs:5]: Please add the correct namespace for the file: `namespace AbstractClasses`

File: Assets/Scripts/AbstractClasses/Movable.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    [Tooltip("The speed to rotate with")]
+    [SerializeField]
+    private int rotationSpeed;
+
+    [Tooltip("The maximum difference in degrees between look direction and target direction in order to be facing the target.")]
+    [SerializeField]
+    private int rotationTolerance;
+
+
+    protected NavMeshAgent _navMeshAgent;
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/AbstractClasses/Movable.cs]:
```suggestion
    protected NavMeshAgent NavMeshAgent;
```
Naming conventions ;)

File: Assets/Scripts/AbstractClasses/Damageable.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+public abstract class Damageable : MonoBehaviour
+{
+    // ------temp for hit animation------
+    [Tooltip("The time the object should be marked as hit after being hit")]
+    [SerializeField]
+    private float hitMarkTime;
+
+    [Tooltip("The MeshRenderer of the graphics object of the player")]
+    [SerializeField]
+    private MeshRenderer gameObjectRenderer;
```
Comment (DavidPgl) [inline Assets/Scripts/AbstractClasses/Damageable.cs]: If you use "RequireComponent" I would convert this to a private field and get the component on awake

File: Assets/Scripts/ControlEffects/ClickEffect.cs [hunk 5]
```diff
@@ -1,29 +1,26 @@
 using UnityEngine;
 
-namespace ControlEffects
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/ControlEffects/ClickEffect.cs:3]: Please re-add the namespace 😅

File: Assets/Scripts/Entity/Enemy/EnemyController.cs [hunk 6]
```diff
@@ -1,65 +1,69 @@
...
-        
-        // component references
-        private NavMeshAgent _agent;
 
+        // component references
         private WanderAI _wanderAI;
         private AttackLogic _attackLogic;
 
         // hold player object to check distance
+        private GameObject _player;
```
Comment (DavidPgl) [inline Assets/Scripts/Entity/Enemy/EnemyController.cs]: Not needed anymore

File: Assets/Scripts/Entity/Enemy/EnemyController.cs [hunk 7]
```diff
@@ -71,11 +75,16 @@ private void Awake()
         /// </summary>
         private void OnEnable()
         {
+            // set target to player
+            _player = PlayerManager.Instance.GetPlayer();
+            if (_player == null)
+                Debug.LogError("No player found.");
+
```
Comment (DavidPgl) [inline Assets/Scripts/Entity/Enemy/EnemyController.cs]: Also not needed

File: Assets/Scripts/Inventory/InventoryController.cs [hunk 8]
```diff
@@ -1,36 +1,31 @@
 ﻿using System.Collections.Generic;
-using Inventory;
 using UnityEngine;
 
-namespace Inventory
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Inventory/InventoryController.cs:5]: Please also re-add the namespace for this file :)

File: Assets/Scripts/Inventory/Item.cs [hunk 9]
```diff
@@ -1,19 +1,16 @@
 ﻿using UnityEngine;
 
-namespace Inventory
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Inventory/Item.cs:3]: Please also re-add the namespace for this file :)

File: Assets/Scripts/Entity/Enemy/EnemyController.cs [hunk 10]
```diff
@@ -170,9 +140,8 @@ private void StartChasing(GameObject target)
         /// </summary>
         private void StopChasing()
         {
-            targetFinder.Targets.RemoveAt(0);
```
Comment (DavidPgl) [inline Assets/Scripts/Entity/Enemy/EnemyController.cs:173]: StopChasing is called when the target ran away successfully, right? This line is needed and can you change/update the summary?

**Review by Kaes3kuch3n — CHANGES_REQUESTED** [review]
Code itself looks fine, but there's a lot of stuff regarding our project conventions that has to be fixed 😬

**Review by DavidPgl — CHANGES_REQUESTED** [review]

File: Assets/Scripts/AbstractClasses/Damageable.cs (hunk 4 above)
Comment (SamuelllG) [reply Assets/Scripts/AbstractClasses/Damageable.cs]: haha okay, you told me to make it SerializeField

File: Assets/Scripts/AbstractClasses/Movable.cs (hunk 3 above)
Comment (SamuelllG) [reply Assets/Scripts/AbstractClasses/Movable.cs]: didn't know a protected var (which is like private for each child) should be written like that

File: Assets/Scripts/AbstractClasses/Damageable.cs (hunk 4 above)
Comment (SamuelllG) [reply Assets/Scripts/AbstractClasses/Damageable.cs]: I deleted requireComponent

File: Assets/Scripts/Entity/Enemy/EnemyController.cs (hunk 10 above)
Comment (SamuelllG) [reply Assets/Scripts/Entity/Enemy/EnemyController.cs:173]: good you saw that, I didn't see it and so didn't merge, thanks!

**Review by SamuelllG — COMMENTED** [review]
sorry for the whole bunch of namespace shit, most of it was deleted at the merge and I forgot some as well!

**Comment by SamuelllG** [comment]
merged again

File: Assets/Scripts/Entity/Enemy/EnemyController.cs (hunk 10 above)
Comment (SamuelllG) [reply Assets/Scripts/Entity/Enemy/EnemyController.cs:173]: forgot to update summary -.- I'm putting it in another PR

File: Assets/Scripts/AbstractClasses/Damageable.cs (hunk 4 above)
Comment (DavidPgl) [reply Assets/Scripts/AbstractClasses/Damageable.cs]: SerializedField -> Not on same GameObject
RequireComponent + Awake() { GetComponent } -> On same GameObject

**Review by DavidPgl — APPROVED** [review]

**Review by Kaes3kuch3n — APPROVED** [review]

**Pull request merged**